File: Core/Responses/Words/GetWordsResponse.cs
using OhMyWord.Data.Models;
using OhMyWord.Data.Services;

namespace OhMyWord.Core.Responses.Words;

public class GetWordsResponse
{
    public int Offset { get; init; }
    public int Limit { get; init; }
    public int Total { get; init; }
    public string Filter { get; init; } = String.Empty;
    public GetWordsOrderBy OrderBy { get; init; }
    public SortDirection Direction { get; init; }
    public IEnumerable<WordResponse> Words { get; init; } = Enumerable.Empty<WordResponse>();
}

File: FrontEnd/src/app/models/responses/get-words.response.ts
import { GetWordsOrderBy } from '../enums/get-words-order-by.enum';
import { SortDirection } from '../enums/sort-direction.enum';
import { WordResponse } from './word.response';

export interface GetWordsResponse {
  offset: number;
  limit: number;
  total: number;
  filter: string;
  orderBy: GetWordsOrderBy;
  direction: SortDirection;
  words: WordResponse[];
}
